File: chts-backend/src/main/java/project/chts/springboot/model/Vision.java
package project.chts.springboot.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Column;

@Entity
@Table(name = "vision")

public class Vision {
	
	@Id
	private int child_id;
	
	@Column(name = "Exophthalmos")
	private String exophthalmos;
	
	@Column(name = "Squint_nys")
	private String squintNys;
	
	@Column(name = "Tagmus")
	private String tagmus;
	
	@Column(name = "Other")
	private String other;
	
	@Column(name = "If_Other_Comment")
	private String ifOtherComment;
	
	@Column(name = "Reffered_To_Specialist")
	private String refferedToSpecialist;
	
	@Column(name = "colorBlind")
	private String colorBlind;
	
	@Column(name = "vision_With_Glasses")
	private String visionWithGlasses;
	
	@Column(name = "vision_Without_Glasses")
	private String visionWithoutGlasses;
	
	@Column(name = "eye_Muscle_Control")
	private String eyeMuscleControl;
	

	public Vision() {
		super();
		// TODO Auto-generated constructor stub
	}
	
	public Vision(int child_id, String exophthalmos, String squintNys, String tagmus, String other,
			String ifOtherComment, String refferedToSpecialist, String colorBlind, String visionWithGlasses,
			String visionWithoutGlasses, String eyeMuscleControl) {
		super();
		this.child_id = child_id;
		this.exophthalmos = exophthalmos;
		this.squintNys = squintNys;
		this.tagmus = tagmus;
		this.other = other;
		this.ifOtherComment = ifOtherComment;
		this.refferedToSpecialist = refferedToSpecialist;
		this.colorBlind = colorBlind;
		this.visionWithGlasses = visionWithGlasses;
		this.visionWithoutGlasses = visionWithoutGlasses;
		this.eyeMuscleControl = eyeMuscleControl;
	}




	public int getChild_id() {
		return child_id;
	}

	public void setChild_id(int child_id) {
		this.child_id = child_id;
	}


	public String getIfOtherComment() {
		return ifOtherComment;
	}

	public void setIfOtherComment(String ifOtherComment) {
		this.ifOtherComment = ifOtherComment;
	}

	public String getRefferedToSpecialist() {
		return refferedToSpecialist;
	}

	public void setRefferedToSpecialist(String refferedToSpecialist) {
		this.refferedToSpecialist = refferedToSpecialist;
	}


	public String getExophthalmos() {
		return exophthalmos;
	}


	public void setExophthalmos(String exophthalmos) {
		this.exophthalmos = exophthalmos;
	}


	public String getSquintNys() {
		return squintNys;
	}


	public void setSquintNys(String squintNys) {
		this.squintNys = squintNys;
	}


	public String getTagmus() {
		return tagmus;
	}


	public void setTagmus(String tagmus) {
		this.tagmus = tagmus;
	}


	public String getOther() {
		return other;
	}


	public void setOther(String other) {
		this.other = other;
	}


	public String getColorBlind() {
		return colorBlind;
	}


	public void setColorBlind(String colorBlind) {
		this.colorBlind = colorBlind;
	}


	public String getVisionWithGlasses() {
		return visionWithGlasses;
	}


	public void setVisionWithGlasses(String visionWithGlasses) {
		this.visionWithGlasses = visionWithGlasses;
	}


	public String getVisionWithoutGlasses() {
		return visionWithoutGlasses;
	}


	public void setVisionWithoutGlasses(String visionWithoutGlasses) {
		this.visionWithoutGlasses = visionWithoutGlasses;
	}


	public String getEyeMuscleControl() {
		return eyeMuscleControl;
	}


	public void setEyeMuscleControl(String eyeMuscleControl) {
		this.eyeMuscleControl = eyeMuscleControl;
	}
	
	
	

}

File: project/src/Components/Vision.js
import './App.css';
import React, {useState, useEffect} from 'react';
import TemporaryDrawer from './navbar';
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import Paper from '@mui/material/Paper';
import Checkbox from '@mui/material/Checkbox';
import FormGroup from '@mui/material/FormGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import FormControl from '@mui/material/FormControl';
import FormLabel from '@mui/material/FormLabel';
import ScreeningNav from './ScreeningNav';
import Radio from '@mui/material/Radio';
import RadioGroup from '@mui/material/RadioGroup';
import { useNavigate, useParams } from 'react-router-dom';
import VisionService from '../Services/VisionService';
import Stack from '@mui/material/Stack';


const Vision = () => {

  const {child_id} = useParams();    
  const [exophthalmos,  setExophthalmos] = useState('');
  const [squintNys,  setSquintNys] = useState('');
  const [tagmus,  setTagmus] = useState('');
  const [other,  setOther] = useState('');
  const [ifOtherComment,  setIfOtherComment] = useState('');
  const [refferedToSpecialist,  setRefferedToSpecialist] = useState('');
  const [colorBlind,  setColorBlind] = useState('');
  const [visionWithGlasses,  setVisionWithGlasses] = useState('');
  const [visionWithoutGlasses,  setVisionWithoutGlasses] = useState('');
  const [eyeMuscleControl,  setEyeMuscleControl] = useState('');
 
  const navigate = useNavigate();

  useEffect(() => {

    VisionService.getChildByID(child_id).then((response) => {
      setExophthalmos(response.data.exophthalmos)
      setSquintNys(response.data.squintNys)
      setTagmus(response.data.tagmus)
      setOther(response.data.other)
      setIfOtherComment(response.data.ifOtherComment)
      setRefferedToSpecialist(response.data.refferedToSpecialist)
      setColorBlind(response.data.colorBlind)
      setVisionWithGlasses(response.data.visionWithGlasses)
      setVisionWithoutGlasses(response.data.visionWithoutGlasses)
      setEyeMuscleControl(response.data.eyeMuscleControl)
    })  
},[])

const saveChildInfo = (e) => {
  e.preventDefault();

  const ChildInfo = {child_id, exophthalmos, squintNys, tagmus, other, ifOtherComment, refferedToSpecialist, colorBlind, visionWithGlasses, visionWithoutGlasses, eyeMuscleControl};

  VisionService.UpdateChild(child_id, ChildInfo).then((response) => {
    console.log(response.data);
    navigate(`/Vision/${child_id}`);
     }).catch(error => {
    console.log(error);
  })   
}
 

  return (
    <div>
    
    <TemporaryDrawer/>
    <ScreeningNav child_id ={child_id}/>
    
    <div>
    <h4 className='page-screeninghead'>Vision Screening</h4>
    <Paper className="form-add">
    <Box
       component="form"
         sx={{
        '& > :not(style)': { m: 1, width: '60ch' },
         }}
           noValidate
          autoComplete="off"
          >

      <FormControl>
      <FormLabel id="demo-row-radio-buttons-group-label">Exophthalmos</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        name = "Exophthalmos" value={exophthalmos} onChange={(e) => setExophthalmos(e.target.value)}>
        <FormControlLabel value={'Yes'} control={<Radio />} label="Yes" />
        <FormControlLabel value= {'No'} control={<Radio />} label="No" />
        <Stack  component="form" sx={{ width: '25ch', }} spacing={2} noValidate autoComplete="off">
        </Stack>
        </RadioGroup>
      </FormControl>
      <FormControl>
      <FormLabel id="demo-row-radio-buttons-group-label">Squint</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        value= {squintNys}  name = "squintNys" onChange={(e) => setSquintNys(e.target.value)} >
        <FormControlLabel value={'Yes'} control={<Radio />} label="Yes" />
        <FormControlLabel value= {'No'} control={<Radio />} label="No" />
        <Stack  component="form" sx={{ width: '25ch', }} spacing={2} noValidate autoComplete="off">
        </Stack>
        </RadioGroup>
      </FormControl>
      <FormControl>
      <FormLabel id="demo-row-radio-buttons-group-label">Nystagmus</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        value={tagmus}  name = "Tagmus" onChange={(e) => setTagmus(e.target.value)} >
        <FormControlLabel value={'Yes'} control={<Radio />} label="Yes" />
        <FormControlLabel value= {'No'} control={<Radio />} label="No" />
        <Stack  component="form" sx={{ width: '25ch', }} spacing={2} noValidate autoComplete="off">
        </Stack>
        </RadioGroup>
      </FormControl>
      <FormControl>
      <FormLabel id="demo-row-radio-buttons-group-label">Color blindness</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        value={colorBlind}  name = "colorBlind" onChange={(e) => setColorBlind(e.target.value)} >
        <FormControlLabel value={'Yes'} control={<Radio />} label="Yes" />
        <FormControlLabel value= {'No'} control={<Radio />} label="No" />
        <Stack  component="form" sx={{ width: '25ch', }} spacing={2} noValidate autoComplete="off">
        </Stack>
        </RadioGroup>
      </FormControl>
      <FormControl>
      <FormLabel >Vision With Glasses</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        name="visionWithGlasses" value={visionWithGlasses} onChange={(e) => setVisionWithGlasses(e.target.value)} >
        <FormControlLabel value={'Good'} control={<Radio />} label="Good" />
        <FormControlLabel value={'Fair'} control={<Radio />} label="Fair" />
        <FormControlLabel value={'Poor'} control={<Radio />} label="Poor" />

      </RadioGroup>
      </FormControl>
      <FormControl>
      <FormLabel >Vision Without Glasses</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        name="visionWithoutGlasses" value={visionWithoutGlasses} onChange={(e) => setVisionWithoutGlasses(e.target.value)} >
        <FormControlLabel value={'Good'} control={<Radio />} label="Good" />
        <FormControlLabel value={'Fair'} control={<Radio />} label="Fair" />
        <FormControlLabel value={'Poor'} control={<Radio />} label="Poor" />

      </RadioGroup>
      </FormControl>
      <FormControl>
      <FormLabel >Eye Muscle Control</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        name="eyeMuscleControl" value={eyeMuscleControl} onChange={(e) => setEyeMuscleControl(e.target.value)} >
        <FormControlLabel value={'Good'} control={<Radio />} label="Good" />
        <FormControlLabel value={'Fair'} control={<Radio />} label="Fair" />
        <FormControlLabel value={'Poor'} control={<Radio />} label="Poor" />

      </RadioGroup>
      </FormControl>
      <FormControl>
      <FormLabel id="demo-row-radio-buttons-group-label">Reffered To Specialist</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        name="refferedToSpecialist" value={refferedToSpecialist} onChange={(e) => setRefferedToSpecialist(e.target.value)}>
        <FormControlLabel value={'Yes'} control={<Radio />} label="Yes" />
        <FormControlLabel value={'No'} control={<Radio />} label="No" />  
      </RadioGroup>
      </FormControl>
        </Box>
        <Box
       component="form"
         sx={{
        '& > :not(style)': { m: 2, width: '138ch' },
         }}
           noValidate
          autoComplete="off">
        <TextField value ={other} name = "other" onChange={(e) => setOther(e.target.value)} label="Vision Screening Comment" variant="outlined" />
      <TextField value ={ifOtherComment} name = "ifOtherComment" onChange={(e) => setIfOtherComment(e.target.value)} label="Comments if any" variant="outlined" />
      </Box>
        </Paper>
    
      <button type="submit" onClick ={(e) => saveChildInfo(e)} className="btn btn-primary save-button">Save</button>
    </div>

    </div>
  );
}

export default Vision;